### PR TITLE
LF-3910: Fix side menu on the left for desktop and tablet views

### PR DESCRIPTION
**Pull request by SayakaOno**
**Description**

- create `SlideMenu` container
- modify `App` component to add `SlideMenu` (it will not slide anymore, but I think I should rename it in another PR since github wouldn't show the diffs if I renamed... I failed to do `git mv` and suffered for a while to get it fixed)
- update NavBar height
- update `NavBar` component to use `SlideMenu` container instead of `SlideMenu` component. (some of NavBar's state are moved to its grandchild component `SlideMenu`)
- create `useIsFarmSelected` hook
- adjust map's width
- adjust stories
- move styles written with makeStyles that I had to update to scss files

NOTE:
- side menu for tablet view will be broken until LF-3912 is completed (let me know if you think I should do better...)
- snackbar's style is currently broken ~~and map's drawer menu needs Loïc's input~~(no change will be made on the map's drawer). To unblock others, I will address these issues later in a separate PR.

Jira link: https://lite-farm.atlassian.net/browse/LF-3910

**Type of change**

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

**How Has This Been Tested?**

- [ ] Passes test case
- [x] UI components visually reviewed on desktop view
- [x] UI components visually reviewed on mobile view
- [ ] Other (please explain)

**Checklist:**

- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [x] The precommit and linting ran successfully
- [ ] I have added or updated language tags for text that's part of the UI
- [ ] I have added "MISSING" for all new language tags to languages I don't speak
- [ ] I have added [the GNU General Public License](https://lite-farm.atlassian.net/l/cp/BT0Dd7WW) to all new files

### CODE REVIEW

File: packages/webapp/src/components/Navigation/NavBar/slideMenu/index.jsx [hunk 1]
```diff
@@ -1,101 +1,82 @@
...
-  const classes = useStyles();
   const handleClick = (link) => {
     history.push(link);
-    closeDrawer();
+    closeDrawer?.();
   };
-  const isAdmin = useSelector(isAdminSelector);
+
   return (
     <div role="presentation" className={classes.container}>
```
Comment (Duncan-Brain) [inline packages/webapp/src/components/Navigation/NavBar/slideMenu/index.jsx:24]: Still uses the {classes.container} here.

**Review by Duncan-Brain — APPROVED** [review]
Menu is staying fixed for me! 

I left a few small inline comments for consideration.

**Comment by Duncan-Brain** [comment]
Just wondering if this is what you meant about side menu for table view being broken. Where the calculation is using max-width versus viewport width. (Using yours as a base for my nav PR so that why I am still maybe finding things haha)

https://github.com/LiteFarmOrg/LiteFarm/assets/30075973/3381ec83-1bcc-4800-a478-eb0d9265bfed

File: packages/webapp/src/App.jsx [hunk 2]
```diff
@@ -13,66 +13,40 @@
...
+import SlideMenu from './containers/Navigation/SlideMenu';
+import styles from './styles.module.scss';
 
 function App() {
-  const classes = useStyles();
   return (
-    <>
-      <div className={clsx(classes.container, classes.defaultHeight, classes.webkitHeight)}>
+    <div className={clsx(styles.container)}>
+      <Suspense fallback={null}>
```
[the remark this answers is not in the document]
Comment (SayakaOno) [reply packages/webapp/src/App.jsx:30]: The child component seems to be waiting `history`, it complains without `<Suspence />`...
`fallback` should be an empty `SlideMenu` I guess...? I will do more research and testing!
@antsgar Any feedback would be appreciated!

File: packages/webapp/src/components/Navigation/NavBar/slideMenu/index.jsx (hunk 1 above)
Comment (SayakaOno) [reply packages/webapp/src/components/Navigation/NavBar/slideMenu/index.jsx:24]: If I'm not mistaken, that's the proper way of taking styles from its parent component...?
(I noticed propType is missing, so I will add it. Thank you!)

File: packages/webapp/src/hooks/useIsFarmSelected.jsx [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+ */
+
+import { useMemo } from 'react';
+import { useSelector } from 'react-redux';
+import { chooseFarmFlowSelector } from '../containers/ChooseFarm/chooseFarmFlowSlice';
+import { userFarmSelector } from '../containers/userFarmSlice';
+import { isAuthenticated } from '../util/jwt';
+
+const useIsFarmSelected = () => {
+  const farm = useSelector(userFarmSelector);
```
[the remark this answers is not in the document]
Comment (SayakaOno) [reply packages/webapp/src/hooks/useIsFarmSelected.jsx:23]: Thank you, I will update!

**Review by antsgar — APPROVED** [review]
Looks good, great work! 👏 

Is it expected that the menu collapsing (not being fixed) on mobile would not work after this PR and be handled in an upcoming one?

**Review by SayakaOno — COMMENTED** [review]
@Duncan-Brain Thank you for reviewing!

> Just wondering if this is what you meant about side menu for tablet view being broken. Where the calculation is using max-width versus viewport width. (Using yours as a base for my nav PR so that why I am still maybe finding things haha)

Yes, that's correct!

**Comment by SayakaOno** [comment]
@antsgar 
Thank you for reviewing!
Yes, for mobile view, I left it as is so that Joyce will be able to use the hamburger icon to trigger the drawer menu. It is going to be handled in https://lite-farm.atlassian.net/browse/LF-3911 :)

**Comment by antsgar** [comment]
@SayakaOno I tried removing the `Suspense` element and I didn't get any errors, can you describe what you're seeing on your end?

**Comment by SayakaOno** [comment]
@antsgar I get a white screen with an error message:
Screenshot 2023-12-12 at 8 36 14 AM

**Comment by antsgar** [comment]
@SayakaOno weird I didn't get that error! I think it's okay to leave the extra suspense for now, seems related to the children components having Suspense

**Review by antsgar — APPROVED** [review]

File: packages/webapp/src/App.jsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (Duncan-Brain) [reply packages/webapp/src/App.jsx:30]: Ahh so thats why, yeah I am having suspense issues too and had to wrap navbar in it or it won't work. Getting the same errors I don't get why history is stalling. Will probably use the same method until a skeleton component is made. Thanks!

**Review by Duncan-Brain — APPROVED** [review]
Looks good!

**Pull request merged**